packages/acceptance-tests/pkg-tests-specs/sources/commands/config.test.js

```diff
@@ -38,7 +39,10 @@ function cleanupPlainOutput(output, path) {
   output = output.replace(new RegExp(path, `g`), FAKE_WORKSPACE_ROOT);
 
   // replace the generated registry server URL with a constant
-  output = output.replace(/^(\s*npmRegistryServer.*?)(['"]?)http:\/\/localhost:\d+\2\s*$/mg, `$1$2${FAKE_REGISTRY_URL}$2`);
+  output = output.replace(/http:\/\/localhost:\d+/g, FAKE_REGISTRY_URL);
+
+  // replace the default global folder with a constant
+  output = output.replace(/[^"]+(\/.local\/share\/yarn\/modern)/g, `${FAKE_USER_ROOT}$1`);
```

Comment:
The global folder is quite different on Windows machines (or machines with XDG installed:

```
function getDefaultGlobalFolder() {
  if (process.platform === `win32`) {
    const folder = NodeFS.toPortablePath(process.env.LOCALAPPDATA || win32.join(homedir(), 'AppData', 'Local'));
    return posix.resolve(folder);
  } else if (process.env.XDG_DATA_HOME) {
    return posix.resolve(process.env.XDG_DATA_HOME, 'yarn/modern');
  } else {
    return posix.resolve(homedir(), `.local/share/yarn/modern`);
  }
}
```

I think that Windows path is incorrect btw: shouldn't we put this in a `yarn/modern` subfolder instead of directly inside `%USERPROFILE%AppData/Local`?
By ensuring the default global folder ends with `yarn/modern` we can use a regex like `/(?<=")[^"]+yarn\/modern(?=")/g`

Reply:
Well spotted, fixed 👍